inject/android.js

```diff
@@ -16,10 +25,43 @@ function initCode () {
         platform: processedConfig.platform
     })
 
+    const messageSecret = processedConfig.messageSecret
+    const messageCallback = processedConfig.messageCallback
+    const messageInterface = processedConfig.messageInterface
+
+    const wrappedUpdate = ((providedSecret, ...args) => {
+        if (providedSecret === messageSecret) {
+            update(...args)
+        }
+    }).bind()
+
+    Object.defineProperty(window, messageCallback, {
+        value: wrappedUpdate
+    })
+
     init(processedConfig)
 
-    // Not supported:
-    // update(message)
+    window.addEventListener('sendMessageProxy' + messageSecret, event => {
+        event.stopImmediatePropagation()
+
+        if (!(event instanceof CustomEvent) || !event?.detail) {
+            return console.warn('no details in sendMessage proxy', event)
+        }
+
+        const messageType = event.detail?.messageType
+        if (!allowedMessages.includes(messageType)) {
+            return console.warn('Ignoring invalid sendMessage messageType', messageType)
+        }
+
+        const message = {
+            type: messageType,
+            options: event.detail?.options,
+            secret: messageSecret
+        }
+        const stringifiedArgs = JSON.stringify(message)
```

Comment:
Is this step required? If so, is it possible to at least avoid passing the secret into `JSON.stringify()`? The website might have messed with that function, so if possible I'd like to avoid giving it the secret.

Reply:
I've sent the secret as a separate argument in process 👍